src/_imaging.c

```diff
@@ -3646,11 +3646,49 @@ _getattr_mode(ImagingObject *self, void *closure) {
     return PyUnicode_FromString(self->image->mode);
 }
 
+static int
+_setattr_mode(ImagingObject *self, PyObject *value, void *closure) {
+    if (value == NULL) {
+        self->image->mode[0] = '\0';
+        return 0;
+    }
+
+    const char *mode = PyUnicode_AsUTF8(value);
+    if (mode == NULL) {
+        return -1;
+    }
+    if (strlen(mode) >= IMAGING_MODE_LENGTH) {
+        PyErr_SetString(PyExc_ValueError, "given mode name is too long");
+        return -1;
+    }
+
+    strcpy(self->image->mode, mode);
+    return 0;
+}
+
 static PyObject *
 _getattr_size(ImagingObject *self, void *closure) {
     return Py_BuildValue("ii", self->image->xsize, self->image->ysize);
 }
 
+static int
+_setattr_size(ImagingObject *self, PyObject *value, void *closure) {
+    if (value == NULL) {
+        self->image->xsize = 0;
+        self->image->ysize = 0;
+        return 0;
+    }
+
+    int xsize, ysize;
+    if (!PyArg_ParseTuple(value, "ii", &xsize, &ysize)) {
+        return -1;
+    }
+
+    self->image->xsize = xsize;
+    self->image->ysize = ysize;
```

Comment:
Why we want to do this? It will just break image

Reply:
So that if `Image.size` is changed, `Image.im.size` will have the same value. Another option could be that if `Image.size` is changed, and `Image.im.size` doesn't match, it raises an exception.